src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -514,7 +514,7 @@ private static function initPlaceholderFormatters(): array
                 $display = str_repeat($bar->getBarCharacter(), $completeBars);
                 if ($completeBars < $bar->getBarWidth()) {
                     $emptyBars = $bar->getBarWidth() - $completeBars - Helper::strlenWithoutDecoration($output->getFormatter(), $bar->getProgressCharacter());
-                    $display .= $bar->getProgressCharacter().str_repeat($bar->getEmptyBarCharacter(), $emptyBars);
```

Comment:
I agree the issue is deeper, there is a confusion in the component between "width" and "length".
`Helper::strlen()` deals with "width", and there is no helper to deal with "length", while it would be needed: a single simple-width character might be enough to replace a single double-width character. To be confirmed of course.

Reply:
See this PR #40695

Am I correct?